int/api/cmd/read_only_call_sc.go

```diff
@@ -0,0 +1,69 @@
+package cmd
+
+import (
+	"encoding/base64"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/massalabs/station/api/swagger/server/models"
+	"github.com/massalabs/station/api/swagger/server/restapi/operations"
+	"github.com/massalabs/station/int/config"
+	"github.com/massalabs/station/pkg/node"
+	sendOperation "github.com/massalabs/station/pkg/node/sendoperation"
+	"github.com/massalabs/station/pkg/wallet"
+)
+
+func NewReadOnlyCallSCHandler(config *config.NetworkInfos) operations.CmdReadOnlyCallSCHandler {
+	return &ReadOnlyCallSC{networkInfos: config}
+}
+
+type ReadOnlyCallSC struct {
+	networkInfos *config.NetworkInfos
+}
+
+func (e *ReadOnlyCallSC) Handle(params operations.CmdReadOnlyCallSCParams) middleware.Responder {
+	args, err := base64.StdEncoding.DecodeString(params.Body.Args)
+	if err != nil {
+		return operations.NewCmdReadOnlyCallSCUnprocessableEntity().
+			WithPayload(
+				&models.Error{
+					Code:    errorInvalidArgs,
+					Message: err.Error(),
+				})
+	}
+
+	acc, err := wallet.Fetch(params.Body.Nickname)
+	if err != nil {
+		return operations.NewCmdReadOnlyCallSCBadRequest().WithPayload(
+			&models.Error{
+				Code:    errorInvalidNickname,
+				Message: "Error during wallet fetch: " + err.Error(),
+			})
+	}
+
+	if params.Body.Fee == "" {
+		params.Body.Fee = "0"
+	}
+
+	coins, errResponse := amountToString(params.Body.Coins, uint64(0))
+	if errResponse != nil {
+		return errResponse
+	}
+
+	result, err := sendOperation.ReadOnlyCallSC(
+		params.Body.At,
+		params.Body.Name,
+		args,
+		coins,
+		string(params.Body.Fee),
```

Comment:
If `Fee` is already of type `string`, why make it as a string again ? 🤔

Reply:
Fee is of type Amount which is type string but golang don't want implicit cast